test/meta/mtk_problem_inputs.jl

```diff
@@ -0,0 +1,243 @@
+    base_sol = solve(base_ssprob, DynamicSS(Tsit5()))
+    base_eprob = EnsembleProblem(base_ssprob)
+    base_esol = solve(base_eprob, DynamicSS(Tsit5()); trajectories = 2)
+
+    # Simulates problems for all input types, checking that identical solutions are found.
+    for u0 in u0_alts, p in p_alts
+        ssprob = remake(base_ssprob; u0, p)
+        @test base_sol == solve(ssprob, DynamicSS(Tsit5()))
+        eprob = remake(base_eprob; u0, p)
+        @test base_esol == solve(eprob, DynamicSS(Tsit5()); trajectories = 2)
+    end
+end
+
+
+### Checks Errors On Faulty Inputs ###
+
+# Checks various erroneous problem inputs, ensuring that these throw errors.
+let
+    # Declares the model.
+    rn = @reaction_network begin
+        (k1,k2), X1 <--> X2
+    end
+    @unpack k1, k2, X1, X2 = rn
+    t = default_t()
+    @species X3(t)
+    @parameters k3
+
+    # Declares valid initial conditions and parameter values
+    u0_valid = [X1 => 1, X2 => 2]
+    ps_valid = [k1 => 0.5, k2 => 0.1]
+
+    # Declares invalid initial conditions and parameters. This includes both cases where values are
+    # missing, or additional ones are given. Includes vector/Tuple/Dict forms.
+    u0s_invalid = [
+        # Missing a value.
+        [X1 => 1],
+        [rn.X1 => 1],
+        [:X1 => 1],
+        Dict([X1 => 1]),
+        Dict([rn.X1 => 1]),
+        Dict([:X1 => 1]),
+        (X1 => 1),
+        (rn.X1 => 1),
+        (:X1 => 1),
+        # Contain an additional value.
+        [X1 => 1, X2 => 2, X3 => 3],
+        [:X1 => 1, :X2 => 2, :X3 => 3],
+        Dict([X1 => 1, X2 => 2, X3 => 3]),
+        Dict([:X1 => 1, :X2 => 2, :X3 => 3]),
+        (X1 => 1, X2 => 2, X3 => 3),
+        (:X1 => 1, :X2 => 2, :X3 => 3)
+    ]
+    ps_invalid = [
+        # Missing a value.
+        [k1 => 1.0],
+        [rn.k1 => 1.0],
+        [:k1 => 1.0],
+        Dict([k1 => 1.0]),
+        Dict([rn.k1 => 1.0]),
+        Dict([:k1 => 1.0]),
+        (k1 => 1.0),
+        (rn.k1 => 1.0),
+        (:k1 => 1.0),
+        # Contain an additional value.
+        [k1 => 1.0, k2 => 2.0, k3 => 3.0],
+        [:k1 => 1.0, :k2 => 2.0, :k3 => 3.0],
+        Dict([k1 => 1.0, k2 => 2.0, k3 => 3.0]),
+        Dict([:k1 => 1.0, :k2 => 2.0, :k3 => 3.0]),
+        (k1 => 1.0, k2 => 2.0, k3 => 3.0),
+        (:k1 => 1.0, :k2 => 2.0, :k3 => 3.0)
+    ]
+
+    # Loops through all potential parameter sets, checking their inputs yield errors.
+    for ps in [ps_valid; ps_invalid], u0 in [u0_valid; u0s_invalid]
+        # Handles problems with/without tspan separately. Special check ensuring that valid inputs passes.
+        for XProblem in [ODEProblem, SDEProblem, DiscreteProblem]
+            if (ps == ps_valid) && (u0 == u0_valid)
+                XProblem(rn, u0, (0.0, 1.0), ps); @test true;
+            else
+                # Several of these cases do not throw errors (https://github.com/SciML/ModelingToolkit.jl/issues/2624).
+                @test_broken false
```

Comment:
Basically a hack so that we get test broken results in the log (since some tests in the loop passes and some fail, cannot just use `test_broken`